plugin/keyring_vault/vault_curl.h

```diff
@@ -0,0 +1,52 @@
+#ifndef MYSQL_VAULT_CURL_H
+#define MYSQL_VAULT_CURL_H
+
+#include <my_global.h>
+#include <curl/curl.h>
```

Comment:
I am not sure whether pimpl is the right thing here. IMHO OK to have a curl dependency for a blahblah_curl.h file, it's not like code modularity would be greatly enhanced or re-compilation times reduced, yet there'd be one more class to maintain.